app/Traits/GetInvoices.php

```diff
@@ -59,24 +51,14 @@ public function getWeeklyInvoices(Admin $admin) {
 	// Return array of integers of this month's invoices
 	public function getMonthlyInvoices(Admin $admin) {
 
-		$invoices_array = [];
-		$invoices = $admin->invoices;
-		$now = Carbon::now();
-		$startOfMonth = $now->startOfMonth();
-
-		for($i = 1; $i <= $now->daysInMonth; $i++) {
-			$invoices_array[$i] = 0;
-		}
+		$invoices_array = array_fill(1, now()->daysInMonth, 0);
+		$invoices = $admin->invoices()
```

Comment:
I've adjusted the query so instead of retrieving all invoices for an admin (you may want this to be all invoices in general for any user), it will only grab the ones within the current month.  This then provides us with two real benefits:
1 - We are letting the database do the heavy lifting for filtering the data (it's faster than PHP is)
2 - We are preventing laravel from importing a lot of extra rows and using more memory

Reply:
I think I see why my code might have been slow... it has to search invocesArray up until [i] for every day of the month.

Reply:
It would only show up in larger data sets but yeah it does have a potential to slow things down.